vm/collector.hpp

```diff
@@ -95,14 +95,13 @@ template <typename TargetGeneration, typename Policy> struct collector {
   }
 
   void trace_partial_objects(cell start, cell card_start, cell card_end) {
-    object* obj = (object*)start;
-    cell end = start + obj->binary_payload_start();
-    start += sizeof(cell);
+    cell *scan_start = (cell*)start + 1;
```

Comment:
Previously ``scan_start = start + sizeof(cell)``, now ``scan_start = start + 1``, is that intentional?